### ISSUE TITLE
Add batch removal

### ISSUE THREAD

**Issue by Lindemann66** [dr2m/activerecord-prunable#6]
At the moment, removal is performed via delete_all or destroy_all, which may not be optimal in the case of a large number of deleting records, along with a complex removal condition. It seems to me that it will be useful to add batch removal - for example, with each_slice.

**Comment by dr2m** [comment]
You're right. It's supposed to be a problem on some occasions.

**Comment by dr2m** [comment]
Resolved in 0.4.0

```ruby
Prunable.prune!(in_batches: true)
Prunable.prune!(batch_size: 100)
```

**Comment by dr2m** [comment]
I've yanked 0.4.0. Use 0.4.4, please.

In this version, you're also able to use batch removal in these ways:
```ruby
class Model
  include ActiveRecord::Prunable
  prune_after 7.days, remove_in_batches: true
end
```

```ruby
class Model
  include ActiveRecord::Prunable
  prune_after 7.days, batch_removal: true
end
```

```ruby
class Model
  include ActiveRecord::Prunable
  prune_after 7.days, remove_in_batches: 50
end
```

```ruby
class Model
  include ActiveRecord::Prunable
  prune_after 7.days, batch_removal: 50
end
```

```ruby
class Model
  include ActiveRecord::Prunable
  scope :prunable, -> { }
  batch_removal
end
```

```ruby
class Model
  include ActiveRecord::Prunable
  scope :prunable, -> { }
  remove_in_batch
end
```

```ruby
class Model
  include ActiveRecord::Prunable
  scope :prunable, -> { }
  batch_removal: 20
end
```

```ruby
class Model
  include ActiveRecord::Prunable
  scope :prunable, -> { }
  remove_in_batches: 20
end
```
